File: esquemas/reportes.py
from pydantic import BaseModel
from datetime import date, datetime

# --- Schema para CREAR un Reporte ---
# Esto es lo que la App (React Native) envía al servidor
class ReporteCreate(BaseModel):
    titulo: str
    descripcion: str | None = None
    fecha_reporte: date
    uuid_cliente: str
    peticion_idempotencia: str | None = None
    id_severidad: int
    id_area: int
# --- Schema para MOSTRAR un Reporte ---
# Esto es lo que el servidor devuelve a la App
class Reporte(BaseModel):
    id_reporte: int
    titulo: str
    descripcion: str | None = None
    fecha_reporte: date
    hora_creado: datetime
    hora_sincronizado: datetime | None = None
    rut: int
    id_severidad: int
    id_area: int
    id_estado_actual: int

    class Config:
        orm_mode = True

File: mobile-app/src/types/reportes.ts
// Tipos básicos para reportes
export interface Reporte {
    id: number;
    titulo: string;
    descripcion: string;
    id_area: number;
    id_severidad: number;
    id_estado_actual: number;
    uuid: string;
    fotos: string[];
    hora_creado: Date;
    hora_sincronizado: Date | null;
    RUT_creador: number; // RUT del trabajador
}

// Tipos para crear un reporte
export interface CreateReportDTO {
    titulo: string;
    descripcion: string;
    fecha_reporte: string; // Formato 'YYYY-MM-DD'
    uuid_cliente: string;
    peticion_idempotencia: string | null;
    id_area: number;
    id_severidad: number;
}

// Tipos para actualizar un reporte
export interface UpdateReportDTO {
    p_nuevo_estado: number;
    p_detalle?: string;
}

// Estado de sincronización
export interface SyncStatus {
    isOnline: boolean;
    lastSync: string | null;
    pendingSyncs: number;
}

// Interfaces para catálogos
export interface Area {
    ID_Area: number;
    Nombre_area: string;
}

export interface Severidad {
    ID_Severidad: number;
    Nombre_severidad: string;
}